addons/pos_self_order/controllers/kiosk_entry.py

```diff
@@ -0,0 +1,92 @@
+# -*- coding: utf-8 -*-
+import werkzeug
+
+from odoo import http
+from odoo.http import request
+from odoo.addons.pos_self_order.controllers.utils import reduce_privilege
+
+
+class PosSelfKiosk(http.Controller):
+    @http.route(["/kiosk/<config_id>", "/kiosk/<config_id>/<path:subpath>"], auth="public", website=True, sitemap=True)
+    def pos_self_order_kiosk_start(self, config_id=None, access_token=None):
+        if not config_id or not access_token:
+            raise werkzeug.exceptions.NotFound()
+
+        lang = request.httprequest.cookies.get('frontend_lang')
+        if lang is not None:
+            language = request.env["res.lang"].sudo().search([("iso_code", "=", lang)], limit=1)
+            if language:
+                lang_code = request.env['res.lang']._lang_get_code(language.code)
+                request.lang = language
+                request.update_context(lang=lang_code)
+
+        pos_config_sudo = request.env["pos.config"].sudo().search([
+            ("id", "=", config_id),
+            ('access_token', '=', access_token)], limit=1)
+
+        if not access_token or not config_id or not pos_config_sudo:
+            raise werkzeug.exceptions.NotFound()
+
+        company = pos_config_sudo.company_id
+        user = pos_config_sudo.current_session_id.user_id
+        pos_config = reduce_privilege(pos_config_sudo, company, user)
+        session_info = request.env["ir.http"].get_frontend_session_info()
+
+        return request.render(
+            'pos_self_order.kiosk_index',
+            {
+                'session_info': {
+                    **session_info,
+                    'currencies': request.env["ir.http"].get_currencies(),
+                    'pos_self_order_data': {
+                        'access_token': pos_config.access_token,
+                        **pos_config._get_self_order_kiosk_data(),
+                    },
+                }
+            }
+        )
+
+    @http.route(
+        "/kiosk/get-bg-image/<int:pos_config_id>/<image>", methods=["GET"], type="http", auth="public"
+    )
+    def pos_self_order_get_bg_image(self, pos_config_id, image):
```

Comment:
not a big leak but if we have the access_token available, we should use it for every controller